aten/src/ATen/core/typeid.h

```diff
@@ -30,6 +30,13 @@
 // fix all of the call-sites, which I want to do later.  So the namespace
 // is not fixed at the moment.
 
+// Make __f16 a fundamental type.
+namespace std {
+template<>
+struct is_fundamental<at::Half> : std::integral_constant<bool, true> {
```

Comment:
I am not sure this is defined behavior, but I guess since Caffe2 has been doing it for a while, should be OK.

Reply:
Specializing a template in namespace `std` for a user-defined type is fine. `std::integral_constant<bool, true>` here should just be `std::true_type`